### PR TITLE
handle odd returns from getting project BGP config

### PR DESCRIPTION
**Pull request by deitch**

### CODE REVIEW

File: packet/bgp.go [hunk 1]
```diff
@@ -141,10 +141,18 @@ func (b *bgp) enableBGP() error {
...
+	// We treat it as valid config already exists only if ALL of the above is true:
+	// - no error
+	// - bgpConfig struct exists
+	// - bgpConfig struct has non-blank ID
+	// - bgpConfig struct does not have Status=="disabled"
+	if err == nil && bgpConfig != nil && bgpConfig.ID != "" && strings.ToLower(bgpConfig.Status) != "disabled" {
 		return nil
 	}
 
+	// we did not have a calid one, so created it
```
Comment (displague) [inline packet/bgp.go:155]: typo here

image

Comment (displague) [reply packet/bgp.go:155]: "so created it" .. and another one.

Comment (deitch) [reply packet/bgp.go:155]: Haha. Nice catch. Fixed.

File: packet/bgp.go [hunk 2]
```diff
@@ -141,10 +141,18 @@ func (b *bgp) enableBGP() error {
...
 	// if we already have a config, just return
-	if err == nil || bgpConfig != nil {
+	// we need some extra handling logic because the API always returns 200, even if
+	// not BGP config is in place.
+	// We treat it as valid config already exists only if ALL of the above is true:
+	// - no error
+	// - bgpConfig struct exists
+	// - bgpConfig struct has non-blank ID
+	// - bgpConfig struct does not have Status=="disabled"
+	if err == nil && bgpConfig != nil && bgpConfig.ID != "" && strings.ToLower(bgpConfig.Status) != "disabled" {
```
Comment (displague) [inline packet/bgp.go:151]: Noting that the possible values for `bgpConfig.Status` are enabled, disabled, and requested.
This set of comparisons looks good.

**Review by displague — APPROVED** [review]

File: packet/bgp.go (hunk 2 above)
Comment (displague) [reply packet/bgp.go:151]: We may want to narrow down the range of errors we are ok with in the future (404s only?).

Comment (deitch) [reply packet/bgp.go:151]: Agreed. We likely will have to redo this anyways once the BGP config endpoints are better structured.

**Pull request merged**